aviary/interface/methods_for_level2.py

```diff
@@ -732,6 +738,213 @@ def run_aviary_problem(
+            # compatibility with being passed int for verbosity
+            verbosity = Verbosity(verbosity)
+        else:
+            verbosity = self.verbosity  # defaults to BRIEF
+
+        # Checks if the sizing mission has run successfully.
+        # If the problem has not run successfully, then we do not run the payload/range function.
+        if self.problem_ran_successfully:
+            # Off-design missions do not currently work for GASP masses or missions.
+            mass_method = self.aviary_inputs.get_val(Settings.MASS_METHOD)
+            equations_of_motion = self.aviary_inputs.get_val(Settings.EQUATIONS_OF_MOTION)
+            # Checks to determine that both the mass and mission methods are set to FLOPS and Height Energy.
+            if (
+                mass_method == LegacyCode.FLOPS
+                and equations_of_motion is EquationsOfMotion.HEIGHT_ENERGY
+            ):
+                # Ensure proper transfer of json files.
+                self.save_sizing_to_json(json_filename='payload_range_sizing.json')
+
+                # make a copy of the phase_info to avoid modifying the original.
+                phase_info = self.model.phase_info
+                phase_info['pre_mission'] = self.model.pre_mission_info
+                phase_info['post_mission'] = self.model.post_mission_info
+                # This checks if the 'cruise' phase exists, then automatically adjusts duration bounds of the cruise stage
+                # to allow the optimizer to arrive at a local maxim for the max_fuel_plus_payload and the ferry ranges.
+                if phase_info['cruise']:
+                    min_duration = phase_info['cruise']['user_options']['time_duration_bounds'][0][
+                        0
+                    ]
+                    max_duration = phase_info['cruise']['user_options']['time_duration_bounds'][0][
+                        1
+                    ]
+                    cruise_units = phase_info['cruise']['user_options']['time_duration_bounds'][1]
+
+                    # Simply doubling the amount of time the optimizer is allowed to stay in the cruise phase, as well as ensure cruise is optimized
+                    phase_info['cruise']['user_options'].update(
+                        {'time_duration_bounds': ((min_duration, 2 * max_duration), cruise_units)}
+                    )
+
+                # point 1 along the y axis (range=0)
+                payload_1 = float(self.get_val(Aircraft.CrewPayload.TOTAL_PAYLOAD_MASS)[0])
+                range_1 = 0
+
+                # point 2, sizing mission which is assumed to be the point of max payload + fuel on the payload and range diagram
+                payload_2 = payload_1
+                range_2 = float(self.get_val(Mission.Summary.RANGE)[0])
+
+                # check if fuel capacity does not exceed sizing mission design gross mass
+                gross_mass = float(self.get_val(Mission.Summary.GROSS_MASS)[0])
```

Comment:
```suggestion
                gross_mass = float(self.get_val(Mission.Design.GROSS_MASS)[0])
```
If you run payload-range after running a different off-design mission you might pull the wrong mass here if you don't specifically call for design? Just covering an edge case